diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1f03954..3dfb015 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -97,14 +97,12 @@ add_custom_command(
 
 add_library("ayatana-ido3-0.4" SHARED ${SOURCES})
 set_target_properties("ayatana-ido3-0.4" PROPERTIES VERSION 0.0.0 SOVERSION 0)
-target_link_libraries("ayatana-ido3-0.4" ${PROJECT_DEPS_LIBRARIES} m)
+target_link_libraries("ayatana-ido3-0.4" ${PROJECT_DEPS_LINK_LIBRARIES} m)
 target_include_directories("ayatana-ido3-0.4" PUBLIC ${PROJECT_DEPS_INCLUDE_DIRS})
 target_include_directories("ayatana-ido3-0.4" PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
 target_include_directories("ayatana-ido3-0.4" PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_definitions("ayatana-ido3-0.4" PUBLIC G_LOG_DOMAIN="IDO")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libayatana-ido3-0.4.so" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libayatana-ido3-0.4.so.0" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libayatana-ido3-0.4.so.0.0.0" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
+install(TARGETS "ayatana-ido3-0.4" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
 
 # AyatanaIdo3-0.4.gir
 
